### PR TITLE
chore(dal): Adding integration tests for creating attribute funcs for props and sockets

### PR DESCRIPTION
**Pull request by stack72**
@nickgerace this is the code we were just working on!

### CODE REVIEW

**Review by nickgerace — CHANGES_REQUESTED** [review]

File: lib/dal/src/schema/variant.rs [hunk 1]
```diff
@@ -1138,6 +1138,24 @@ impl SchemaVariant {
...
+                    .get_node_weight_by_id(func_id)
+                    .await?
+                    .get_func_node_weight()?;
+
+                if node_weight.func_kind() == FuncKind::Attribute {
+                    let func = Func::get_by_id_or_error(ctx, func_id).await?;
+                    all_funcs.push(func);
+                }
+            }
+        }
```
[the remark this answers is not in the document]
Comment (nickgerace) [reply lib/dal/src/schema/variant.rs:1157]: (this is why I requested changes)

File: lib/dal/tests/integration_test/func/authoring/create_func.rs [hunk 2]
```diff
@@ -193,6 +199,197 @@ async fn create_codegen_with_schema_variant(ctx: &mut DalContext) {
...
+//
+//     ctx.update_visibility_and_snapshot_to_visibility(head_change_set)
+//         .await
+//         .expect("Unable to go back to HEAD");
+//
+//     let head_func = Func::find_by_name(ctx, func_name.clone())
+//         .await
+//         .expect("Unable to get a func");
+//     assert!(head_func.is_none());
+// }
```
[the remark this answers is not in the document]
Comment (stack72) [reply lib/dal/tests/integration_test/func/authoring/create_func.rs]: :ok

File: lib/dal/src/func/authoring/create.rs [hunk 3]
```diff
@@ -202,9 +202,13 @@ async fn create_attribute_func(
...
+                    // right now, we don't allow overiding identity
+                    // See create_attribute_with_socket and create_attribute_with_prop tests for
+                    // examples of where this breaks
                     let func_id = AttributePrototype::func_id(ctx, ap).await?;
                     if let Some(func) = Func::get_by_id(ctx, func_id).await? {
-                        if !Func::is_dynamic_for_name_string(func.name.as_str()) {
+                        if Func::is_dynamic_for_name_string(func.name.as_str()) {
                             return Err(FuncAuthoringError::AttributePrototypeAlreadySetByFunc(
                                 kind.to_string(),
                             ));
```
[the remark this answers is not in the document]
Comment (stack72) [reply lib/dal/src/func/authoring/create.rs:214]: I have left this for now - but I want to have a conversation about it soon :)

**Comment by stack72** [comment]
bors try

**Comment by stack72** [comment]
bors merge

**Pull request merged**